### ISSUE TITLE
tuple assignment in let block

### ISSUE THREAD

**Issue by xdavidliu** [JuliaLang/julia#16647]
It is currently not possible to perform tuple assignment in a let block. For example,

```
let x=1, y=2
    x+y
end
```

works but not

```
let (x,y)=(1,2)
    x+y
end
```

While I am fine with always using the first option, the fact that the second option doesn't work makes it awkward when I have a function that returns a tuple, such as

```
function foo()
   (1, 2)
end
```

and I have to do

```
let tup = foo()
    tup[1] + tup[2]
end
```

instead of 

```
let (x,y) = foo()
    x+y
end
```

**Comment by JeffBezanson** [comment]
I believe this is fixed in 0.5.

**Comment by yuyichao** [comment]
Ref https://github.com/JuliaLang/julia/pull/15036
Test already included in https://github.com/JuliaLang/julia/blob/15626c30ae59eb76549a52dbf8bed798b3c44595/test/core.jl#L3646
Unlikely going to be backported...
